### PR TITLE
Update draft-ietf-ipsecme-rfc8229bis.xml

### PR DESCRIPTION
**Pull request by smyslov**
WGLC comments from Paul Wouters and Tero Kivinen

### CODE REVIEW

File: draft-ietf-ipsecme-rfc8229bis.xml [hunk 1]
```diff
@@ -46,8 +46,8 @@
 
       <abstract>
         <t> This document describes a method to transport Internet Key Exchange
-        Protocol (IKE) and IPsec packets over a TCP connection for traversing
-        network middleboxes that may block IKE negotiation over UDP.  This
+        Protocol (IKE) and IPsec packets over a TCP connection in case routers or network middleboxes 
+        do not handle UDP, so that IKE negotiation over UDP would fail.  This
```
Comment (tfpauly) [inline draft-ietf-ipsecme-rfc8229bis.xml]: I find this a bit harder to understand — it's not about boxes that don't handle UDP in general, it's about ones that block IKE over UDP specifically.

Since this is text from the original, I'd prefer to leave as is.

```suggestion
        Protocol (IKE) and IPsec packets over a TCP connection for traversing
        network middleboxes that may block IKE negotiation over UDP.  This
```

File: draft-ietf-ipsecme-rfc8229bis.xml [hunk 2]
```diff
@@ -564,20 +564,34 @@
             </t>
             <t>if the Responder chooses to send Cookie request (possibly along
             with Puzzle request), then the TCP connection that the IKE_SA_INIT
-            request message was received over SHOULD be closed, so that the
-            Responder remains stateless at least until the Cookie (or Puzzle
-            Solution) is returned. Note that if this TCP connection is closed,
-			the Responder MUST NOT include the Initiator's TCP port into the Cookie
-			calculation (*), since the Cookie will be returned over a new
+            request message was received over SHOULD be closed after the Responder sends its reply
+            and no repeated requests are received within some short period of time (*)
```
Comment (tfpauly) [inline draft-ietf-ipsecme-rfc8229bis.xml]: I find these (*) references hard to follow, particularly with large amounts of text. Why not just have a sub-section with a proper section reference?

File: draft-ietf-ipsecme-rfc8229bis.xml [hunk 3]
```diff
@@ -692,13 +715,14 @@
 
       <section title="Interaction with IKEv2 Extensions" anchor="extensions" >
         <section title="MOBIKE Protocol" anchor="sect-8">
-          <t>MOBIKE protocol, that allows IKEv2 SA to migrate between IP
+          <t>The MOBIKE protocol, that allows SAs to migrate between IP
```
Comment (tfpauly) [inline draft-ietf-ipsecme-rfc8229bis.xml]:
```suggestion
          <t>The MOBIKE protocol, which allows SAs to migrate between IP
```

File: draft-ietf-ipsecme-rfc8229bis.xml [hunk 4]
```diff
@@ -724,7 +748,7 @@
 		  presence of NATs.
           </t>
 
-          <t>MOBIKE protocol defined the NO_NATS_ALLOWED notification that can be
+          <t>MOBIKE protocol defines the NO_NATS_ALLOWED notification that can be
```
Comment (tfpauly) [inline draft-ietf-ipsecme-rfc8229bis.xml]:
```suggestion
          <t>The MOBIKE protocol defines the NO_NATS_ALLOWED notification that can be
```

**Review by tfpauly — CHANGES_REQUESTED** [review]

File: draft-ietf-ipsecme-rfc8229bis.xml (hunk 1 above)
Comment (smyslov) [reply draft-ietf-ipsecme-rfc8229bis.xml]: It was Paul Wouters' suggestion to make text more neutral. Original text he thinks sounds like we design a circumvention protocol to bypass filtering policies which IETF he thinks is not mandated to design. I have no problem with the original text, but I think that Paul's comment should be addressed somehow. Can you reply to his review? https://mailarchive.ietf.org/arch/msg/ipsec/8DB-i0afGewqFAhsgGBRg_6vAHM/

File: draft-ietf-ipsecme-rfc8229bis.xml (hunk 2 above)
Comment (smyslov) [reply draft-ietf-ipsecme-rfc8229bis.xml]: I created a new subsection discussing selecting time interval and referenced it in the text. The text about TCP ports in Cookie calculation is left under (*). Please review.

File: draft-ietf-ipsecme-rfc8229bis.xml (hunk 1 above)
Comment (tfpauly) [reply draft-ietf-ipsecme-rfc8229bis.xml]: Sure, I can reply via email, but I’d prefer not to change this now.

**Comment by smyslov** [comment]
Are you OK with the other changes? Can I merge the PR?

File: draft-ietf-ipsecme-rfc8229bis.xml [hunk 5]
```diff
@@ -564,11 +564,11 @@
...
-            request message was received over SHOULD be closed, so that the
-            Responder remains stateless at least until the Cookie (or Puzzle
-            Solution) is returned. Note that if this TCP connection is closed,
-			the Responder MUST NOT include the Initiator's TCP port into the Cookie
-			calculation (*), since the Cookie will be returned over a new
+            request message was received over SHOULD be closed after the Responder sends its reply
+            and no repeated requests are received within some short period of time 
+            to keep the Responder stateless (see <xref target="tradeoff" />). Note that the Responder MUST NOT 
+            include the Initiator's TCP port into the Cookie
+			calculation (*), since the Cookie may be returned over a new
```
Comment (tfpauly) [inline draft-ietf-ipsecme-rfc8229bis.xml]: Let's avoid using words that can be normative:

```suggestion
			calculation (*), since the Cookie can be returned over a new
```

File: draft-ietf-ipsecme-rfc8229bis.xml [hunk 6]
```diff
@@ -589,6 +589,19 @@
...
+          <section title="Statelessness versus Delay of SA Establishment" anchor="tradeoff" >
+            <t>
+            There is a trade-off in choosing the period of time after which
+            TCP connection is closed. If it is too short, then the proper Initiator
+            which repeats its request would need to re-establish the TCP connection
+            introducing additional delay. On the other hand, if it is too long, then 
+            the Responder's resources would be wasted in case the Initiator never comes back.
+            This document doesn't specify the duration of time, because it doesn't affect interoperability, 
+            but it is believed that 5-10 seconds is a good compromise. Note also, that if the Responder requests 
+            the Initiator to solve a puzzle, then the Responder may estimate how long it would take the Initiator
```
Comment (tfpauly) [inline draft-ietf-ipsecme-rfc8229bis.xml]:
```suggestion
            the Initiator to solve a puzzle, then the Responder can estimate how long it would take the Initiator 
```

**Review by tfpauly — APPROVED** [review]
